addons/misra.py

```diff
@@ -840,13 +841,19 @@ def isFloatCounterInWhileLoop(whileToken):
     token = whileBodyStart
     while token != whileBodyStart.link:
         token = token.next
-        for counterToken in counterTokens:
-            if not counterToken.valueType or not counterToken.valueType.isFloat():
-                continue
-            if token.isAssignmentOp and token.astOperand1.str == counterToken.str:
+        if token.str not in tok_str:
```

Comment:
hmm the old code only matched the token string but that is unfortunate.

we should not determine that `f` is a loop counter here:
```
void foo(void)
{
    int i = 0;
    float f = 0;
    while (f < 10 && i < 10) {
        float f = 0; f = f + i;
        i++;
    }
}
```